embedded-vertica/src/test/java/com/playtika/test/vertica/EmbeddedVerticaBootstrapConfigurationTest.java

```diff
@@ -0,0 +1,82 @@
+package com.playtika.test.vertica;
+
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import javax.sql.DataSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Slf4j
+@SpringBootTest(
+        classes = EmbeddedVerticaBootstrapConfigurationTest.TestConfiguration.class,
+        properties = {
+                "spring.profiles.active=enabled",
+                "embedded.vertica.enabled=true"
+        }
+)
+class EmbeddedVerticaBootstrapConfigurationTest {
+    @Autowired
+    ConfigurableListableBeanFactory beanFactory;
+
+    @Autowired
+    JdbcTemplate verticaJdbcTemplate;
+
+    @Autowired
+    ConfigurableEnvironment environment;
+
+    @Test
+    public void shouldConnectToVertica() {
+        assertThat(verticaJdbcTemplate.queryForObject("SELECT version()", String.class)).contains("Vertica Analytic Database");
+    }
+
+    @Test
+    public void propertiesAreAvailable() {
+        assertThat(environment.getProperty("embedded.vertica.port")).isNotEmpty();
+        assertThat(environment.getProperty("embedded.vertica.host")).isNotEmpty();
+        assertThat(environment.getProperty("embedded.vertica.database")).isNotEmpty();
+        assertThat(environment.getProperty("embedded.vertica.user")).isNotEmpty();
+        assertThat(environment.getProperty("embedded.vertica.password")).isNotNull();
+    }
+
+    @EnableAutoConfiguration
+    @Configuration
+    static class TestConfiguration {
+        @Value("${spring.datasource.url}")
+        String jdbcUrl;
+
+        @Value("${spring.datasource.username}")
+        String user;
+
+        @Value("${spring.datasource.password}")
+        String password;
+
+        @Value("${spring.datasource.driver-class-name}")
+        String driverClassName;
+
+        @Bean
+        public DataSource verticaDataSource() {
```

Comment:
Autoconfiguration works fine on the local machine but during the HIT build it fails whit an error
`Error:  com.playtika.test.vertica.EmbeddedVerticaBootstrapConfigurationTest.propertiesAreAvailable  Time elapsed: 0.006 s  <<< ERROR!
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'com.playtika.test.vertica.EmbeddedVerticaBootstrapConfigurationTest': Unsatisfied dependency expressed through field 'jdbcTemplate'; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.springframework.jdbc.core.JdbcTemplate' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true)}`
Due to this I created the bean manually.

Reply:
so where is garantee that in real spring application it will work, I feel something is missing in test to boostrap spring datasource, and we need to figure out it prior of merging this PR

in the same time the goal is to have vertical integration, but since this progect is dedicated exactly to spring boot, I feel that we need to check that integration should be smooth for clients.

Reply:
this was due to missed dependency. Fixed in next PR